server/src/main/java/org/opensearch/action/get/TransportGetAction.java

```diff
@@ -90,14 +93,21 @@ protected boolean resolveIndex(GetRequest request) {
 
     @Override
     protected ShardIterator shards(ClusterState state, InternalRequest request) {
+        String preference = request.request().preference();
```

Comment:
Can we also add unit test to ensure preference is changed to `primary` for segment replication enabled indices ?

Reply:
(1) is covered by tests in SegmentReplicationIT testing realtime get and multi get requests. (2) is covered by existing tests in GetActionIT.